python_hb_report/collector.py

```diff
@@ -0,0 +1,374 @@
+
+		scsi_info = utillib.do_command(['lsscsi'])
+		msg = msg +'\n'+ scsi_info
+
+		pci_info = utillib.do_command(['lspci'])
+		msg = msg +'\n'+ pci_info
+
+		mount_info = utillib.do_command(['mount'])
+		msg = msg +'\n' + mount_info
+
+		#df can block, run in background, allow for 5 seconds
+		df_pro = subprocess.Popen(['df'],stderr = subprocess.STDOUT,stdout = subprocess.PIPE)
+		timer = threading.Timer(5.0,df_pro.kill)
+		timer.start()
+		df_info = df_pro.communicate()[0]
+		if timer.is_alive():
+			#df exited naturally, cancel timer
+			timer.cancel()
+		msg = msg +'\n'+df_info
+
+		f.write(msg)
+		f.close()
+
+	def pe2dot(self,path):
+		pef = utillib.basename(path)
+		if pef.endswith('.bz2'):
+			dotf = pef[0:len(pef)-4]
+
+		if not len(envir.PTEST):
+			return False
+		try:
+			msg = utillib.do_command([envir.PTEST,'-D','dotf','-x',pef])
+		except:
+			utillib.debug(envir.PTEST+' faild! ')
+			return
+
+	def dlm_dump(self):
+		output = os.path.join(self.WORKDIR,envir.DLM_DUMP_F)
+		if not utillib.do_which('dlm_tool'):
+			return 
+		dlm_file_info = 'NOTICE - Lockspace overview:\n'	
+		dlm_file_info = dlm_file_info + utillib.do_command(['dlm_tool','ls'])
+
+		dlm_pro = subprocess.Popen(['dlm_tool','ls'],stdout = subprocess.PIPE,stderr = subprocess.STDOUT)
+		grep_pro = subprocess.Popen(['grep','name'],stdin = dlm_pro.stdout,stdout = subprocess.PIPE,stderr = subprocess.STDOUT)
+
+		dlm_list = grep_pro.communicate()[0].split('\n')
+
+		for d in dlm_list:
+			strd = d.split()
+			for X,N in strd:
+				dlm_file_info = dlm_file_info+'\n'+'NOTICE - Lockspace '+N
+				dlm_file_info = dlm_file_info +utillib.do_command(['dlm_tool','lock_dump',N])
+
+		dlm_file_info = dlm_file_info +'\nNOTICE - Lockspace histoty:\n'
+		dlm_file_info = dlm_file_info +utillib.do_command(['dlm_tool','dump'])
+
+		utillib.writefile(output,dlm_file_info)
+
+
+
+
+	def getpeinputs(self,workdir):
+		i = 0
+
+		utillib.debug('looking for PE files in'+envir.PE_STATE_DIR)
+		flist = utillib.find_files(envir.PE_STATE_DIR.split())
+		grep_pro = subprocess.Popen(['grep','-v',"[.]last$"],stdin = subprocess.PIPE,stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+		flist = grep_pro.communicate(' '.join(flist))[0].split()
+		
+		if len(flist):
+			filename = utillib.basename(envir.PE_STATE_DIR)
+			pengine_dir = os.path.join(workdir,filename)
+			os.mkdir(pengine_dir)
+			for f in flist:
+#				os.symlink(f,os.path.join(pengine_dir,utillib.basename(f)))
+				os.symlink(f,pengine_dir)
+				utillib.do_command(['ln','-s',f,pengine_dir])
+				i = i + 1
+			utillib.debug('found '+str(i)+' pengine input files in '+envir.PE_STATE_DIR)
+
+		if i >= 20:
+			for f in flist:
+				if not self.skip_lvl(1):
+					path = os.path.join(workdir,utillib.basename(envir.PE_STATE_DIR))
+					path = os.path.join(path,utillib.basename(f))
+					self.pe2dot(path)
+		else:
+			utillib.debug('too many PE inputs to create dot files')
+
+	def touch_DC_if_dc(self):
+		dc = utillib.do_command(['crmadmin','-D'])
```

Comment:
sometimes utillib.do_command, and sometimes subprocess.Popen, is there any reason to do so?